hbase-common/src/main/java/org/apache/hadoop/hbase/CellComparatorImpl.java

```diff
@@ -387,4 +409,23 @@ public Comparator getSimpleComparator() {
   public Comparator getSimpleComparator() {
     return new BBKVComparator(this);
   }
+
+  /**
+   * Utility method that makes a guess at comparator to use based off passed tableName.
+   * Use in extreme when no comparator specified.
+   * @return CellComparator to use going off the <code>tableName</code> passed.
+   */
+  public static CellComparator getCellComparator(TableName tableName) {
+    return getCellComparator(tableName.toBytes());
```

Comment:
Can we just use TableName.isMetaTableName(TableName) here?  Why to have the indirection of toBytes and then compare bytes?

Reply:
Need both. In actual filecontext, it hosts tablename as bytes only -- not as a TableName object. The TableName.toBytes doesn't actually make bytes. TN itself hosts the name in bytes.

Maybe I should be clearer in a comment that no new arrays are being made in this code?